src/devices/cpu/z80/z80.cpp

```diff
@@ -621,34 +619,35 @@ void z80_device::device_start()
-	state_add(Z80_B,           "B",         B).noshow();
-	state_add(Z80_C,           "C",         C).noshow();
-	state_add(Z80_D,           "D",         D).noshow();
-	state_add(Z80_E,           "E",         E).noshow();
-	state_add(Z80_H,           "H",         H).noshow();
-	state_add(Z80_L,           "L",         L).noshow();
-	state_add(Z80_AF,          "AF",        AF);
-	state_add(Z80_BC,          "BC",        BC);
-	state_add(Z80_DE,          "DE",        DE);
-	state_add(Z80_HL,          "HL",        HL);
-	state_add(Z80_IX,          "IX",        IX);
-	state_add(Z80_IY,          "IY",        IY);
-	state_add(Z80_AF2,         "AF2",       m_af2.w);
-	state_add(Z80_BC2,         "BC2",       m_bc2.w);
-	state_add(Z80_DE2,         "DE2",       m_de2.w);
-	state_add(Z80_HL2,         "HL2",       m_hl2.w);
-	state_add(Z80_WZ,          "WZ",        WZ);
-	state_add(Z80_R,           "R",         m_rtemp).callimport().callexport();
-	state_add(Z80_I,           "I",         m_i);
-	state_add(Z80_IM,          "IM",        m_im).mask(0x3);
-	state_add(Z80_IFF1,        "IFF1",      m_iff1).mask(0x1);
-	state_add(Z80_IFF2,        "IFF2",      m_iff2).mask(0x1);
-	state_add(Z80_HALT,        "HALT",      m_halt).mask(0x1);
+	state_add(STATE_GENPC,          "PC",        m_pc.w).callimport();
+	state_add(STATE_GENPCBASE,      "CURPC",     m_prvpc.w).callimport().noshow();
+	state_add(Z80_SP,               "SP",        SP);
+	state_add(STATE_GENFLAGS,       "GENFLAGS",  F).noshow().formatstr("%8s");
+	state_add(Z80_A,                "A",         A).noshow();
+	state_add(Z80_B,                "B",         B).noshow();
+	state_add(Z80_C,                "C",         C).noshow();
+	state_add(Z80_D,                "D",         D).noshow();
+	state_add(Z80_E,                "E",         E).noshow();
+	state_add(Z80_H,                "H",         H).noshow();
+	state_add(Z80_L,                "L",         L).noshow();
+	state_add(Z80_AF,               "AF",        AF);
+	state_add(Z80_BC,               "BC",        BC);
+	state_add(Z80_DE,               "DE",        DE);
+	state_add(Z80_HL,               "HL",        HL);
+	state_add(Z80_IX,               "IX",        IX);
+	state_add(Z80_IY,               "IY",        IY);
+	state_add(Z80_AF2,              "AF2",       m_af2.w);
+	state_add(Z80_BC2,              "BC2",       m_bc2.w);
+	state_add(Z80_DE2,              "DE2",       m_de2.w);
+	state_add(Z80_HL2,              "HL2",       m_hl2.w);
+	state_add(Z80_WZ,               "WZ",        WZ);
+	state_add(Z80_R,                "R",         m_rtemp).callimport().callexport();
+	state_add(Z80_I,                "I",         m_i);
+	state_add(Z80_IM,               "IM",        m_im).mask(0x3);
+	state_add(Z80_IFF1,             "IFF1",      m_iff1).mask(0x1);
+	state_add(Z80_IFF2,             "IFF2",      m_iff2).mask(0x1);
+	state_add(Z80_HALT,             "HALT",      m_halt).mask(0x1);
+	state_add(Z80_INPUT_LINE_BUSRQ, "BUSRQ",     m_busrq_state).mask(0x1);
```

Comment:
I'm not sure about adding input pins to state_add. That aside, there's a conflict here:
Z80_INPUT_LINE_BUSRQ = 2
Z80_A = 2

Reply:
Ah, I'll remove then